git-changebar: fix compatibility with libgit2 0.28 API

maxice8 opened the pull request:
Namely rename the deprecated git_buf_free to git_buf_dispose and change
giterr calls to git_error.

elextr commented:
This needs to select which name to use depending on the version of libgit the plugin is being built against.

maxice8 commented:
Ok they have been guarded under

```cpp
#if LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR >= 28
/* new api */
#else 
/* old api */
#endif 
```

elextr commented:
According to the [releasse notes](https://github.com/libgit2/libgit2/releases) API removals for 0.28, the old names are being kept for backward compatibility "for the foreseeable future", so this isn't really needed.

b4n commented:
I didn't check the new API yet, but it's probably good to use it.  But I'd rather see a backward compatibility chunk like there's around line 35, and change the code to use the new name.

b4n commented on git-changebar/src/gcb-plugin.c:
```diff
@@ -463,7 +471,11 @@ worker_thread (gpointer data)
+#if LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR >= 28
+          git_buf_dispose (&job->buf);
+#else
+          git_buf_fre (&job->buf);
```
typo reintroducing `git_buf_free()`

maxice8 replied:
fixed

maxice8 commented:
I removed the individual checks and added a block that defines git_buf_dispose to git_buf_free and git_error_last to giterr_last when libgit2 version is under 0.28.0

b4n commented on git-changebar/src/gcb-plugin.c:
```diff
@@ -45,6 +45,10 @@
 #endif
+#if LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR < 28
+#define git_buf_dispose(x) git_buf_free(x)
+#define git_error_last() giterr_last()
```
there's also no need to include arguments as they are identical, it would be enough to do
```c
# define git_buf_dispose  git_buf_free
# define git_error_last   giterr_last
```

maxice8 replied:
Fixed, thanks for the review

maxice8 commented:
:+1:

b4n commented:
> and indeed fixes support for a naive libgit2 build (which is not too nice on their part, but well… might actually be a bug).

It is indeed a libgit2 [bug](https://github.com/libgit2/libgit2/issues/4978) and they are working on a [fix](https://github.com/libgit2/libgit2/pull/4979).
